[14.0][OU-IMP] account: Compute stored function fields by SQL

etobella opened the pull request:
When managing files from account.move, account.move.line, account.payment, account.bank.statement or account.bank.statement.line we should always migrate them by SQL, not using the computed function (it could take hours or, even worse, fail for a Memory failure)

Candidate fields
- [x] account.move
  - [x] sequence_prefix
  - [x] sequence_number
  - [x] payment_state --> Already done on preovious migration
- [x] account.bank.statement
  - [x] sequence_prefix
  - [x] sequence_number
  - [x] is_valid_balance_start
  - [x] previous_statement_id
- [x] account.bank.statement.line
  - [x] amount_residual
  - [x] is_reconciled
- [x] account.move.line
  - [x] matching_number
- [x] account.payment
  - [x] is_internal_transfer
  - [x] destination_account_id
  - [x] is_matched
  - [x] is_reconciled
  - [x] partner_bank_id
- [x] account.partial.reconcile
  - [x] credit_currency_id
  - [x] debit_currency_id

etobella committed:
[OU] account: Compute sequence mixin data by SQL

etobella committed:
[OU] account: fill matching_number by SQL

etobella committed:
[OU] account: The partner must be set by the partner of the journal related

etobella committed:
[OU] account: Set destination_account_id and is_internal_transfer of account payment by SQL

etobella committed:
[OU] account: Set partner_bank_id from account.payment

etobella committed:
[OU] account: Calculate currency fields for account.partial.reconcile

etobella committed:
[OU] account: payment reconcile fields

etobella committed:
[OU] account: Fill reconciliation fields of account

etobella committed:
[OU] account: calculate account_bank_statement fields

chienandalu reviewed (approved):
:+1: Thanks for this!

pedrobaeza committed:
[OU-FIX+IMP] account: Payment data filling optimization

- Don't use account_move, as for draft payment, they are not available,
  and anyway, we can extract the needed data from account_journal.
- Perform only 2 queries for destination_account_id instead of 8, and
  also do it more readable.

pedrobaeza committed:
[OU-FIX] account: Normalize work file format

pedrobaeza committed:
[OU-FIX] account: Clean and finish SQL query for AP reconciliation flags

- It's more expressive and shorter doing JOIN with ON clause.
- is_reconciled flag was not updated.
- Nested CASE for shorting the query and making it more readable.
- Extra: Assign False by default on column creation for avoiding
  NULL/False dualities.
